### ISSUE TITLE
Default browser image icon with title = "Image not found"

### ISSUE THREAD

**Issue by sabina-talipova** [silverstripe/silverstripe-admin#1709]
## Description
ENV: SC PHP8.3 Chrome Installer

I see default browser image icon with title = "Image not found" if I insert image in WYSIWYG editor and image is not published. 

<img width="824" alt="Screenshot_2024-03-13_at_10 33 43_AM (1)" src="https://github.com/silverstripe/silverstripe-admin/assets/87288324/3fbf4ee3-3635-4359-a418-3ef9986c8c5b">

## Steps to reproduce issue
- I insert image in Content field (WYSIWYG editor) on a Page
- I publish the Page
- I can see the image as unauthorized user
- As admin I unpublish the image in CMS "Files" section
- I cannot see the image as unauthorized user, but I see default image icon.

**Comment by GuySartorelli** [comment]
What steps did you take to get into this condition?
I assume you unpublished the image **after** publishing the page?

Have you checked this behaviour locally in `5.1` to be sure it's a regression with Silverstripe CMS?

**Comment by sabina-talipova** [comment]
> What steps did you take to get into this condition?

I follow scenarios that we have in Cucumber Studio. 

> I assume you unpublished the image after publishing the page?

It is one of the step. Please, see this instruction (https://studio.cucumber.io/projects/301855/test-plan/folders/2145967/scenarios/7080337). 

> Have you checked this behaviour locally in 5.1 to be sure it's a regression with Silverstripe CMS?

I'll check, but I think it exists in CMS 5.1 as well, we just ignored it as an issue when did last regression tests. It's not a regression issue. 
I think main question is it expected behaviour, that we show browser image icon like a message "here should be image, but something went wrong" or we should avoid any missing content.

**Comment by GuySartorelli** [comment]
Yeah definitely something we need to decide how to handle - my main point is if it's not a regression we can handle it after 5.2.0, whereas if it's a regression we probably need to prioritise it sooner.

**Comment by maxime-rainville** [comment]
I couldn't replicate this locally. Do we still have this issue in the sandbox?

**Comment by sabina-talipova** [comment]
I tested on 5.2 and 5.1 in my local. Issue exists. 

<img width="384" alt="Screenshot 2024-03-19 at 8 26 59 AM" src="https://github.com/silverstripe/silverstripe-admin/assets/87288324/72d74808-a3ec-4539-82a5-cbbcae0d3f90">

I've updated the description with steps how to reproduce issue.
